File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num = {}
        for i,n in enumerate(nums):
            complement = target - n
            if complement in num:
                return [num[complement],i]
            num[n] =i

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>map;
        int n = nums.size();
        for(int i=0; i<n; i++){
            int comp = target-nums[i];
            if(map.count(comp)){
                return {map[comp],i};
            }
            map[nums[i]] = i;
        }
        return{};
        
    }
};